web-services/model/src/main/java/datawave/webservice/query/model/ModelBean.java

```diff
@@ -109,8 +123,13 @@
     @Path("/list")
     @GZIP
     @Interceptors(ResponseInterceptor.class)
-    public ModelList listModelNames(@QueryParam("modelTableName") @DefaultValue(DEFAULT_MODEL_TABLE_NAME) String modelTableName) {
-        ModelList response = new ModelList();
+    public ModelList listModelNames(@QueryParam("modelTableName") String modelTableName) {
+        
+        if (modelTableName == null) {
+            modelTableName = defaultModelTableName;
+        }
+        
+        ModelList response = new ModelList(modelTableName, dataTablesUri, modelTableName);
```

Comment:
Looks like you're passing `modelTableName` twice here. That doesn't match up with the `ModelList` parameters. The first param should be `jqueryUri`.

Reply:
The modelTable name has multiple points of entry.  On the query string or or dependency injection.  Or if it can't check the modelTable name (the metadata table name) it uses the DEFAULT_MODEL_TABLE_NAME in the ModelBean.java.

Reply:
> 
> 
> The modelTable name has multiple points of entry. On the query string or or dependency injection. Or if it can't check the modelTable name (the metadata table name) it uses the DEFAULT_MODEL_TABLE_NAME in the ModelBean.java.

What does this have to do with my comment? You are still passing the wrong values to the ModelList constructor here.

Reply:
> 
> 
> @brianloss . The second parameter in the ModelList constructor is the path the to the DataTables javascript library and has nothing to do with the model name at all. That is handled by the 'names' hashset which gets used on line 100 to build the table.
> 
> I fixed the misspelling of the third parameter.

You're still missing it. First off, see my other comment about the name of the third parameter. The name does not match its purpose and I would like it changed. Regarding the error, I don't know how to express it more clearly.
Look at the 3 parameters you are passing:

1. modelTableName
2. dataTablesUri
3. modelTableName

Now look at what the constructor takes:

1.  jqueryUri
2. dataTablesUri
3. (misnamed) dataWaveMetadataTableName

Don't you see how passing modelTableName to a parameter that expects to receive the jqueryUri is wrong?